### PR TITLE
Add securityContext settings

### PR DESCRIPTION
**Pull request by xhejtman**
Add option to specify security context for Kubernetes pods

### CODE REVIEW

**Review by martin-kuba — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 1]
```diff
@@ -124,6 +124,17 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>org.yaml</groupId>
+            <artifactId>snakeyaml</artifactId>
+            <version>1.21</version>
```
[the remark this answers is not in the document]
Comment (xhejtman) [reply pom.xml]: fixed

**Comment by xhejtman** [comment]
@lvarin any idea how to fix these build fails?

**Comment by lvarin** [comment]
It is complaining about:

https://github.com/elixir-cloud-aai/tesk-api/blob/3ca56c5c9b1fb9de025c5103ff260e1ae34434fb/src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterTest.java#L87

I am putting the link here, so I can find it later easier.

**Comment by lvarin** [comment]
It took me whole day, but I got the problem. I will add the suggestions to a review, but the gist of it is that:

* The two ` @ContextConfiguration(classes = {io.kubernetes.client.openapi.apis.CoreV1Api.class})` should be removed
* coreAPi must be added to ` TesKubernetesConverterMinimalTest.java`:

```
@MockBean
private CoreV1Api coreApi;
``

File: src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java [hunk 2]
```diff
@@ -58,6 +59,7 @@
 @JsonTest
 @TestPropertySource(locations = {"classpath:application.properties"})
 @EnableConfigurationProperties(TaskmasterEnvProperties.class)
+@ContextConfiguration(classes = {io.kubernetes.client.openapi.apis.CoreV1Api.class})
```
Comment (lvarin) [inline src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java:62]:
```suggestion

```

File: src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java [hunk 3]
```diff
@@ -58,6 +59,7 @@
...
 public class TesKubernetesConverterMinimalTest {
```
Comment (lvarin) [inline src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java:64]:
```suggestion
    @MockBean
    private CoreV1Api coreApi;

```

**Comment by xhejtman** [comment]
I wonder why it worked before merge from master.

**Review by lvarin — CHANGES_REQUESTED** [review]
The suggested changes should fix the tests. Please check if they make sense for you. I have not tested the change in OpenShift, but will do after these changes are committed.

File: src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java [hunk 4]
```diff
@@ -21,6 +21,7 @@
 import org.springframework.context.annotation.Import;
 import org.springframework.core.io.ClassPathResource;
 import org.springframework.core.io.Resource;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.TestPropertySource;
 import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.util.StringUtils;
```
[the remark this answers is not in the document]
Comment (xhejtman) [reply src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java:27]: fixed

File: src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java (hunk 3 above)
Comment (xhejtman) [reply src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java:66]: fixed

File: src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java (hunk 2 above)
Comment (xhejtman) [reply src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterMinimalTest.java]: fixed

File: src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterTest.java [hunk 5]
```diff
@@ -70,7 +71,10 @@
                 "tesk.api.k8s.namespace = test-namespace"
         })
 @EnableConfigurationProperties(TaskmasterEnvProperties.class)
+@ContextConfiguration(classes = {io.kubernetes.client.openapi.apis.CoreV1Api.class})
```
[the remark this answers is not in the document]
Comment (xhejtman) [reply src/test/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverterTest.java]: fixed

**Comment by xhejtman** [comment]
@lvarin thanks a lot! it is fixed now. so can we merge?

**Comment by lvarin** [comment]
I will check today or tomorrow, sorry I had no time yet.

**Review by martin-kuba — APPROVED** [review]

**Review by lvarin — APPROVED** [review]

**Comment by lvarin** [comment]
LGTM, it works out of the box in OpenShift (which is very picky with the security context)

**Pull request merged**